src/Persistence/Sql.php

```diff
@@ -592,11 +592,11 @@ public function action(Model $model, string $type, array $args = [])
                 $this->initQueryConditions($model, $query);
                 $model->hook(self::HOOK_INIT_SELECT_QUERY, [$query, $type]);
 
-                // MSSQL does not support EXISTS everywhere, so wrap in SELECT
-                if ($this->connection instanceof \atk4\dsql\Mssql\Connection) {
-                    return $this->dsql()->expr(
-                        '(select case when exists[] then 1 else 0 end)',
-                        [$query]
+                // MSSQL and Oracle do not support EXISTS everywhere, so wrap in CASE
+                if ($this->connection instanceof \atk4\dsql\Mssql\Connection
```

Comment:
We should move this to Connection specific class code, e.g
```
$this->dsql()->exists($query);
```
And we have specific `\atk4\dsql\Oracle\Expression::exists` method

Reply:
In current design, Expression is not too much too be extended :( as used a lot directly and already further extended.

however, yes, it should be moved, related probably with #690

Reply:
I implemented it similar way in #690 for now but added a @todo for moving it